core/services/job/models.go

```diff
@@ -278,7 +278,21 @@ type JSONConfig map[string]interface{}
 
 // Bytes returns the raw bytes
 func (r JSONConfig) Bytes() []byte {
-	b, _ := json.Marshal(r)
+	var retCopy = make(JSONConfig, len(r))
+	for key, value := range r {
+		copiedVal := value
+		// If the value is a json structure string, unmarshal it to preserve JSON structure
+		// e.g. instead of this {"key":"{\"nestedKey\":{\"nestedValue\":123}}"}
+		//      we want this    {"key":{"nestedKey":{"nestedValue":123}}},
+		if strValue, ok := copiedVal.(string); ok {
+			if object, ok := asObject(strValue); ok {
+				copiedVal = object
+			}
+		}
+		retCopy[key] = copiedVal
+	}
```

Comment:
This seems hacky, why do we need this?

Reply:
This seems like a workaround for incorrectly encoded test data.

Reply:
I couldn't figure out how to embed json into RelayConfig in any way other than
`[relayconfig]
 chainReader=`{...}`
`
And for this string to be properly converted into json it requires this special handling.

Happy to change this if there is a better way

Reply:
I don't understand what you mean in your example - can you link to a test case?

There is a canonical way to escape TOML strings that must be followed, regardless of whether the contents are JSON or anything else. I would write an end-to-end encode/decode test, and let the code inform you of what the proper encoded format is, rather than trying to construct it by hand.

Reply:
By this do you mean storing the configs as toml instead of embedding json into toml? Then it is a question of readability, I thought we wanted to go with json

Reply:
> By this do you mean storing the configs as toml instead of embedding json into toml? 

No, I just mean using the appropriate escaping at each level.

Can you please link to where this is being used in code? We are probably just constructing it wrong.

Reply:
[link](https://github.com/smartcontractkit/chainlink/blob/1163401427b639ed1f6ba8284f8bbabc0dd4f6b2/integration-tests/actions/ocr2_helpers_local.go#L129)

Not sure if escaping is the issue, Json struct just doesn't differentiate between string and json string which was added

Reply:
Something doesn't add up, because there is an asymmetry.

Reply:
Oh I think the test value can just be specified via `json.RawMessage` instead of a plain `string`.

Reply:
Otherwise, we should be using values instead.

Reply:
The original `json.RawMessage` idea wasn't valid, but that PR demonstrates a working pattern now. Need to clean up a few more things though.